analyzer/codechecker_analyzer/analyzers/clangsa/ctu_manager.py

```diff
@@ -120,27 +120,47 @@ def generate_ast(triple_arch, action, source, config, env):
                   cmdstr, err)
 
 
-def func_map_list_src_to_ast(func_src_list):
+def ast_dump_path(source_path):
+    """ AST-dump based analysis uses preprocessed paths, here the path prefix
+    'ast' and the filename suffix '.ast' is added to the name of the original
+    source file. """
+
+    # Normalize path on Windows OS.
+    path = os.path.splitdrive(source_path)[1]
+    # Make relative path out of absolute.
+    path = path[1:] if path[0] == os.sep else path
+    # Prepend path segment, and append filename suffix.
+    return os.path.join("ast", path + ".ast")
+
+
+def func_map_list_src_to_ast(func_src_list, ctu_on_demand):
     """ Turns textual function map list with source files into a
-    function map list with ast files. """
+    mapping from mangled names to mapped paths, which can be absolute paths to
+    the original source files if ctu_on_demand is True, or relative path
+    segments to AST-dump files otherwise. """
 
     func_ast_list = []
     for fn_src_txt in func_src_list:
         dpos = fn_src_txt.find(" ")
         mangled_name = fn_src_txt[0:dpos]
         path = fn_src_txt[dpos + 1:]
-        # Normalize path on windows as well
-        path = os.path.splitdrive(path)[1]
-        # Make relative path out of absolute
-        path = path[1:] if path[0] == os.sep else path
-        ast_path = os.path.join("ast", path + ".ast")
-        func_ast_list.append(mangled_name + " " + ast_path)
+
+        # On-demand analysis does not require any preprocessing on the source
+        # file paths, contrary to AST-dump based.
+        mapped_path = path if ctu_on_demand else ast_dump_path(path)
+
+        func_ast_list.append(mangled_name + " " + mapped_path)
     return func_ast_list
 
 
 def map_functions(triple_arch, action, source, config, env,
                   func_map_cmd, temp_fnmap_folder):
-    """ Generate function map file for the current source. """
+    """ Generate function map file for the current source.
```

Comment:
Do we use the same file name for both type of data? Would there be any conflict if I switch between the two types and the original ctu tries to read up the function map file created for the on-demand?

Reply:
Right now the index format is (usr-name -> file-system-path). In case of dump based (vanilla) ctu the path is something that has an `.ast` postfix. If it has any other postfix, the entry is supposed to be used with on-demand loader.